From e08471fc7f20d3db8b66d6bf708bcab60ca38597 Mon Sep 17 00:00:00 2001
From: Zheng Zengkai <zhengzengkai@huawei.com>
Date: Tue, 25 Jan 2022 20:10:06 +0800
Subject: [PATCH] BMA: Fix pointer cast compile warning in arm32 builds
MIME-Version: 1.0
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: e08471fc7f20d3db8b66d6bf708bcab60ca38597
Modified-by-SEL: No

Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4ETXO
CVE: NA

-----------------------------------------

Fix following build warnings of iBMA driver in arm32 builds:

drivers/net/ethernet/huawei/bma/cdev_veth_drv/virtual_cdev_eth_net.c: In function ‘__make_dmalistbd_b2h_H_2’:
drivers/net/ethernet/huawei/bma/cdev_veth_drv/virtual_cdev_eth_net.c:1266:38: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  pdmalbase_v[i].slow = lower_32_bits((u64)prxtx_queue->pdmalbase_p);
                                      ^
./include/linux/kernel.h:191:34: note: in definition of macro ‘lower_32_bits’
 #define lower_32_bits(n) ((u32)((n) & 0xffffffff))
                                  ^
drivers/net/ethernet/huawei/bma/cdev_veth_drv/virtual_cdev_eth_net.c:1267:37: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
  pdmalbase_v[i].shi = upper_32_bits((u64)prxtx_queue->pdmalbase_p);
                                     ^
./include/linux/kernel.h:185:35: note: in definition of macro ‘upper_32_bits’
 #define upper_32_bits(n) ((u32)(((n) >> 16) >> 16))
                                   ^

Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/huawei/bma/Kconfig | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/huawei/bma/Kconfig b/drivers/net/ethernet/huawei/bma/Kconfig
index 3c4c7b5d9757..cfecd37289a2 100644
--- a/drivers/net/ethernet/huawei/bma/Kconfig
+++ b/drivers/net/ethernet/huawei/bma/Kconfig
@@ -4,7 +4,7 @@
 
 config BMA
 	tristate "Huawei BMA Driver"
-
+	depends on 64BIT
 	help
 		This driver supports Huawei BMA Software. It is used
-		to communication between Huawei BMA and BMC software.
\ No newline at end of file
+		to communication between Huawei BMA and BMC software.
-- 
2.26.2

